pkg/client/rp/relying_party.go

```diff
@@ -249,6 +270,20 @@ func NewRelyingPartyOIDC(ctx context.Context, issuer, clientID, clientSecret, re
 	rp.oauthConfig.Endpoint.AuthStyle = rp.oauthAuthStyle
 	rp.endpoints.Endpoint.AuthStyle = rp.oauthAuthStyle
 
+	if rp.pkce == pkceFromDiscovery {
+		if slices.Contains(
+			discoveryConfiguration.CodeChallengeMethodsSupported,
+			oidc.CodeChallengeMethodPlain,
+		) || slices.Contains(
+			discoveryConfiguration.CodeChallengeMethodsSupported,
+			oidc.CodeChallengeMethodS256,
+		) {
+			rp.pkce = pkceEnabled
```

Comment:
Check for both values in a single iteration by using a `ContainsFunc`

```suggestion
		if slices.ContainsFunc(discoveryConfiguration.CodeChallengeMethodsSupported, func(method oidc.CodeChallengeMethod) bool {
			return method == oidc.CodeChallengeMethodPlain ||
				method == oidc.CodeChallengeMethodS256
		}) {
			rp.pkce = pkceEnabled
```